packages/server/graphql/public/types/Company.ts

```diff
@@ -0,0 +1,204 @@
+    const r = await getRethink()
+    const organizations = await dataLoader.get('organizationsByActiveDomain').load(domain)
+    const orgIds = organizations.map(({id}) => id)
+    const teamsByOrgId = (await dataLoader.get('teamsByOrgIds').loadMany(orgIds)).filter(
+      errorFilter
+    )
+    const teams = teamsByOrgId.flat()
+    const teamIds = teams.map(({id}) => id)
+    if (teamIds.length === 0) return 0
+    return (
+      r
+        .table('NewMeeting')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .filter((row) => row('endedAt').default(null).ne(null))
+        // number of months since unix epoch
+        .merge((row: RValue) => ({
+          epochMonth: row('endedAt').month().add(row('endedAt').year().mul(12))
+        }))
+        .group((row) => [row('teamId'), row('epochMonth')])
+        .count()
+        .ungroup()
+        .map((row) => ({
+          teamId: row('group')(0),
+          epochMonth: row('group')(1)
+        }))
+        .group('teamId')('epochMonth')
+        .ungroup()
+        .map((row) => ({
+          teamId: row('group'),
+          epochMonth: row('reduction'),
+          // epochMonth shifted 1 index position
+          shift: row('reduction')
+            .deleteAt(0)
+            .map((z) => z.add(-1))
+        }))
+        .merge((row: RValue) => ({
+          // 1 if there are 2 consecutive epochMonths next to each other, else 0
+          teamStreak: r
+            .map(row('shift'), row('epochMonth'), (shift, epochMonth) =>
+              r.branch(shift.eq(epochMonth), '1', '0')
+            )
+            .reduce((left, right) => left.add(right).default(''))
+            .default('')
+            // get an array of all the groupings of 1
+            .split('0')
+            .map((val) => val.count())
+            .max()
+            .add(1)
+        }))
+        .max('teamStreak')('teamStreak')
+        .run()
+    )
+  },
+
+  organizations: async ({id: domain}, _args, {dataLoader}) => {
+    const organizations = await dataLoader.get('organizationsByActiveDomain').load(domain)
+    return organizations
+  },
+
+  tier: async ({id: domain}, _args, {dataLoader}) => {
+    const organizations = await dataLoader.get('organizationsByActiveDomain').load(domain)
+    const tiers = organizations.map(({tier}) => tier)
+    if (tiers.includes('enterprise')) return 'enterprise'
+    if (tiers.includes('pro')) return 'pro'
+    return 'personal'
+  },
+
+  userCount: async ({id: domain}, _args, {dataLoader}) => {
+    const organizations = await dataLoader.get('organizationsByActiveDomain').load(domain)
+    const orgIds = organizations.map(({id}) => id)
+    const organizationUsersByOrgId = (
+      await dataLoader.get('organizationUsersByOrgId').loadMany(orgIds)
+    ).filter(errorFilter)
+    const organizationUsers = organizationUsersByOrgId.flat()
+    const userIds = organizationUsers.map((organizationUser) => organizationUser.userId)
```

Comment:
Are you initially creating an array and then transforming it to a set vs initially creating a set because you feel it's more readable?

Reply:
Thanks, that makes sense!